staging/src/k8s.io/component-base/metrics/testutil/metrics.go

```diff
@@ -68,6 +85,19 @@ func NewMetrics() Metrics {
 	return result
 }
 
+// NewMetricsFromSample returns new Metrics which is initialized with Sample.
+func NewMetricsFromSample(kv map[string][]*Sample) Metrics {
```

Comment:
This function is called from test/e2e_node/dynamic_kubelet_config_test.go only.
Why don't we put this function in the file directly?
The following 2 functions also are called in the file only.

I might miss some point.

Reply:
This PR is mainly to remove k/k dependency on prometheus. So move functions used prometheus to component-base/metrics.

Reply:
is there plan to remove the reference from `component-base/metrics` later? Is the benefit of moving it here that all references will be addressed at once?